### DISCUSSION TITLE
Adding search bar to membership template

### DISCUSSION THREAD

**Discussion by ldwIwan** [Budibase/budibase#7803] [Help]
Hi all,

Using the membership template app to create a membership bank of all our members. I've been trying to implement a search bar however, I've been unsuccessful with every online tutorial I follow.  

Would anyone know or have a link to how I can add a basic search bar to filter through members names/emails etc?

Here's the membership template I've been using - https://budibase.com/directories/templates/membership-directory/

Big thank you to all those who respond.

**Comment by mslourens** [comment]
You can add a search input using the following steps:

1. Add a form in the root of the screen and name it Form
![image](https://user-images.githubusercontent.com/1907152/191599271-f6f14704-aee3-46c8-9733-fdaefeb76f3f.png)

2. Add a TextField to the Form component and name it SearchInput
![image](https://user-images.githubusercontent.com/1907152/191599786-9b26ada0-a620-462b-89ae-ca4ac69b7a98.png)

3. Move the dataprovider with the table inside the Form component
4. Select the dataprovider and click the `Define Filters` button
5. Click the `Add Filter` button to add a new filter
6. Select a column you want to use in your search
7. Select `Like` in the compare dropdown
8. Select `Binding` in the value dropdown
9. Enter this binding template in the value textfield: `{{ Form.Fields.SearchInput }}`. It should look like this:
![image](https://user-images.githubusercontent.com/1907152/191599990-05e45345-21ad-4490-85e6-6a90b0058c70.png)

Now whenever you enter text in the search input, the dataprovider gets filtered. If you want to use multiple fields for your search, you will have to define a filter for each column in the dataprovider.

Hope this helps.
